### PR TITLE
Shipper Creating a Load

### PR DESCRIPTION
**Pull request by matzayonc**

### CODE REVIEW

File: programs/protocol/src/data/transport.rs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+#[derive(Debug, Default, PartialEq, BorshSerialize, BorshDeserialize)]
+pub struct TransportDetails {
+    priority: u8,
+    fragility: u8,
+    reserved: [u8; 6], // size is excessive, but required for bytemuck
+}
+
+#[zero_copy]
+#[derive(Debug, Default, PartialEq, BorshSerialize, BorshDeserialize)]
+pub struct TransportDimensions {
```
Comment (tommysr) [inline programs/protocol/src/data/transport.rs:16]: we need more precise dimensions, w x h x d

Comment (tommysr) [reply programs/protocol/src/data/transport.rs:16]: instead of volume

File: programs/protocol/src/data/shipper.rs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+use anchor_lang::prelude::*;
+
+#[account(zero_copy)]
+#[derive(Debug, Default, PartialEq, Eq)]
+pub struct Shipper {
+    pub authority: Pubkey,
+    pub count: u32,
```
Comment (tommysr) [inline programs/protocol/src/data/shipper.rs:7]: what does the count mean?

File: programs/protocol/src/data/transport.rs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+#[account(zero_copy)]
+#[derive(Debug, Default, PartialEq, BorshSerialize, BorshDeserialize)]
+pub struct Transport {
+    pub from: Coordinates,
+    pub to: Coordinates,
+    pub dimensions: TransportDimensions,
+    pub when: u64,
+    pub transport_details: TransportDetails,
+}
```
Comment (tommysr) [inline programs/protocol/src/data/transport.rs:36]: What about more info about "from" and "to" location?
Sometimes its impossible to approach some locations with big trucks.

File: programs/protocol/src/data/transport.rs [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    pub longitude: f32,
+}
+
+#[account(zero_copy)]
+#[derive(Debug, Default, PartialEq, BorshSerialize, BorshDeserialize)]
+pub struct Transport {
+    pub from: Coordinates,
+    pub to: Coordinates,
+    pub dimensions: TransportDimensions,
+    pub when: u64,
```
Comment (tommysr) [inline programs/protocol/src/data/transport.rs:34]: I thought about some range instead of actual value

**Review by tommysr — COMMENTED** [review]
Would be nice to to have some deploy script for populating state, when new structures are created.

File: programs/protocol/src/data/transport.rs (hunk 4 above)
Comment (matzayonc) [reply programs/protocol/src/data/transport.rs:34]: lets assume everyone wants have the shippment out of their hands as soon as possible, and no later than deadline

**Comment by matzayonc** [comment]
> Would be nice to do have some deploy script for populating state, when new structures are created.

lets do it as a separate PR

File: programs/protocol/src/data/transport.rs (hunk 3 above)
Comment (matzayonc) [reply programs/protocol/src/data/transport.rs:39]: let's have it as one of the `ShipmentDetails`, because the whole trip will be limited by this

**Review by tommysr — APPROVED** [review]

**Pull request merged**